src/main/java/cz/centrum/haffner/SimpleTrainingDavid/REST/SimpleTrainingDavidController.java

```diff
@@ -2,24 +2,39 @@
 
 import cz.centrum.haffner.SimpleTrainingDavid.AppServices.KpisInfoService;
 import cz.centrum.haffner.SimpleTrainingDavid.AppServices.MetricsInfoService;
-import cz.centrum.haffner.SimpleTrainingDavid.DataTemplates.KpisInfoSet;
+import cz.centrum.haffner.SimpleTrainingDavid.DataTemplates.KpisInfoDataSet;
+import cz.centrum.haffner.SimpleTrainingDavid.DataTemplates.MetricsInfoDataSet;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class SimpleTrainingDavidController {
 
-        @GetMapping("/{requestedDate}/metrics")   // supposed date format: YYYYMMDD
-        public MetricsInfoService getMetrics(@PathVariable long requestedDate) {
-            MetricsInfoService metricsInfoService = new MetricsInfoService();
-            metricsInfoService.proceedMetricsInfoSet(requestedDate);
-            return metricsInfoService;
-        }
+    @Autowired
+    private MetricsInfoService metricsInfoService;
 
-        @GetMapping("/kpis")
-        public KpisInfoSet getKpis() {
-            KpisInfoService kpisInfoService = new KpisInfoService();
-            return kpisInfoService.proceedKpisInfoSet();
-        }
+    /*
+    @GetMapping("/{requestedDate}/metrics")   // supposed date format: YYYYMMDD
+    public MetricsInfoService getMetrics(@PathVariable long requestedDate) {
+        metricsInfoService.proceedMetricsInfoSet(requestedDate);
+        return metricsInfoService;
+    } */
+
+    @GetMapping("/{requestedDate}/metrics")   // supposed date format: YYYYMMDD
+    public @ResponseBody ResponseEntity<MetricsInfoDataSet> getMetrics(@PathVariable long requestedDate) {
+        MetricsInfoDataSet response = metricsInfoService.proceedMetricsInfoDataSet(requestedDate);
+        return new ResponseEntity<MetricsInfoDataSet>(response, HttpStatus.OK);
```

Comment:
Správně použitá ResponseEntity.
Jen ji použij při vracení výstupu z restu i pro ten druhý GET :-D

Reply:
předěláno ...